src/Components/Alert/Hooks/__tests__/AlertProvider.jest.tsx

```diff
@@ -0,0 +1,121 @@
+import { MockEnvironment, createMockEnvironment } from "relay-test-utils"
+import { useTracking } from "react-tracking"
+import { graphql } from "react-relay"
+
+import { AlertProvider } from "Components/Alert/AlertProvider"
+import { CreateAlertButton } from "Components/Alert/Components/CreateAlertButton"
+import { AlertProviderTestPage } from "Components/Alert/Hooks/__tests__/Utils/AlertProviderTestPage"
+import { useAuthDialog } from "Components/AuthDialog"
+import { setupTestWrapper } from "DevTools/setupTestWrapper"
+import { useSystemContext } from "System/SystemContext"
+import { flushPromiseQueue } from "DevTools/flushPromiseQueue"
+
+jest.unmock("react-relay")
+jest.mock("react-tracking")
+
+jest.mock("Components/AuthDialog/useAuthDialog")
+jest.mock("Utils/Events", () => ({
+  postEvent: jest.fn(),
+}))
+jest.mock("System/useFeatureFlag", () => ({
+  useFeatureFlag: jest.fn(() => true),
+}))
+jest.mock("System/useSystemContext")
+
+describe("AlertProvider", () => {
+  let relayEnv: MockEnvironment = createMockEnvironment()
+  let trackEvent: jest.Mock
+
+  const { getWrapper } = setupTestWrapper({
+    Component: ({ artist }: any) => (
+      <AlertProvider initialCriteria={{ artistIDs: [artist.internalID] }}>
+        <CreateAlertButton />
+      </AlertProvider>
+    ),
+    query: graphql`
+      query AlertProviderTestQuery @raw_response_type @relay_test_operation {
+        artist(id: "artist-id") {
+          internalID
+        }
+      }
+    `,
+  })
+
+  beforeAll(() => {
+    trackEvent = jest.fn()
+    ;(useTracking as jest.Mock).mockImplementation(() => ({ trackEvent }))
```

Comment:
`mockTrackEvent = trackEvent as jest.Mock`
`mockUseSystemContext = ...` 
`mockUseAuthDialog`

Reply:
Thanks! Hopefully [this](https://github.com/artsy/force/pull/13153/commits/9cd7fe1b7b2d2e8c81297250ccd719b234e55fb4) is what you mean. Definitely looks a bit cleaner.